rest/config.go

```diff
@@ -1461,9 +1461,9 @@ func (sc *ServerContext) migrateV30Configs(ctx context.Context) error {
 		if insertErr != nil {
 			if insertErr == base.ErrAlreadyExists {
 				base.DebugfCtx(ctx, base.KeyConfig, "Found legacy config for database %s, but already exists in registry.", base.MD(dbConfig.Name))
-				continue
+			} else {
+				return fmt.Errorf("Error migrating v3.0 config for bucket %s groupID %s: %w", base.MD(bucketName), base.MD(groupID), insertErr)
 			}
-			return fmt.Errorf("Error migrating v3.0 config for bucket %s groupID %s: %w", base.MD(bucketName), base.MD(groupID), insertErr)
 		}
 		removeErr := sc.BootstrapContext.Connection.DeleteMetadataDocument(ctx, bucketName, PersistentConfigKey30(ctx, groupID), legacyCas)
```

Comment:
Is it worth worrying about another node hitting this delete simultaneously, or should we just not worry about the remove error?

An idea might be pass on a KeyNotFoundError, but fail on other errors (SDK timeout)?

Reply:
Considering this further, I don't think we should be terminating iteration through the buckets on an error for a single bucket (particularly since config migration is designed as a best effort, and doesn't block startup).  Going to rework the earlier cases that are returning error to instead log and continue.